client/allocdir/task_dir_linux.go

```diff
@@ -0,0 +1,36 @@
+package allocdir
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	multierror "github.com/hashicorp/go-multierror"
+)
+
+// unmountSpecialDirs unmounts the dev and proc file system from the chroot. No
+// error is returned if the directories do not exist or have already been
+// unmounted.
+func (t *TaskDir) unmountSpecialDirs() error {
+	errs := new(multierror.Error)
+	dev := filepath.Join(t.Dir, "dev")
+	if pathExists(dev) {
```

Comment:
Is there a bit better validation we can do. What if I just create a dev/ or proc/ folder in raw_exec/docker case. Now we may fail unlinking a normal directory

Reply:
Since this is only used in destroying a pre09 task, perhapse we just ignore the errors and TRACE level log them?

Reply:
I think returning an error is fine since it just gets logged. Traced it and realized this function isn't returning the error: https://github.com/hashicorp/nomad/blob/master/client/allocrunner/alloc_runner_hooks.go#L228-L252

Can you fix that in this PR